alerter.cpp

```diff
@@ -5,27 +5,31 @@ int alertFailureCount = 0;
 
 int networkAlertStub(float celcius) {
     std::cout << "ALERT: Temperature is " << celcius << " celcius.\n";
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+   if(celcius>200)
+        return 500;
+    else      
+        return 200;   
+}
+
+float convertFarenehitToCelcius(float farenheit)
+{
+    float celcius;
+    celcius = ((farenheit-32)*5/9);
+    return celcius;
 }
 
 void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
+    float celcius = convertFarenehitToCelcius (farenheit);
     int returnCode = networkAlertStub(celcius);
     if (returnCode != 200) {
-        // non-ok response is not an error! Issues happen in life!
-        // let us keep a count of failures to report
-        // However, this code doesn't count failures!
-        // Add a test below to catch this bug. Alter the stub above, if needed.
         alertFailureCount += 0;
     }
 }
 
 int main() {
     alertInCelcius(400.5);
     alertInCelcius(303.6);
+    assert(alertFailureCount>= 1);

```

Comment:
this is a weak criteria. The count should be absolute as the input. Since we know exactly what is the input, the behavior of the production code should be exact too. Assert exact value.